Utils.py

```diff
@@ -1073,3 +1073,13 @@ def is_iterable_except_str(obj: object) -> TypeGuard[typing.Iterable[typing.Any]
     if isinstance(obj, str):
         return False
     return isinstance(obj, typing.Iterable)
+
+
+class WeakReferencableDict(dict):
```

Comment:
There are alternatives to adding this `WeakReferenceableDict` class.

All that is needed is some wrapper or container class that supports having weak references, and keeps a strong reference to the game data package it is wrapping/containing.

Reply:
Keeping manual count of how many rooms are using each gamepackage and removing that gamepackage from the shared dictionary when the count reaches zero during a room's `__del__`, could be another alternative, but that sounds more complicated to me.